lib/api/unversioned/types.go

```diff
@@ -30,8 +30,8 @@ var (
 //
 // All resource and resource lists embed a TypeMetadata as an anonymous field.
 type TypeMetadata struct {
-	Kind       string `json:"kind" validate:"required"`
-	APIVersion string `json:"apiVersion" validate:"required"`
+	Kind       string `json:"kind"`
+	APIVersion string `json:"apiVersion"`
```

Comment:
Removing the validate here removes a check because some uses would not have the values.  Are there cases where `Kind` and `APIVersion` should be present and tests should be added in explicitly?